### PR TITLE
test: onchain tests to top level bats

### PR DESCRIPTION
**Pull request by sandipndev**
- Create Random user helper
- E2E test onchain stuff to top level

### CODE REVIEW

File: bats/helpers/_common.bash [hunk 1]
```diff
@@ -118,3 +118,11 @@ cache_value() {
 read_value() {
   cat ${CACHE_DIR}/$1
 }
+
+bitcoin_cli() {
```
Comment (bodymindarts) [inline bats/helpers/_common.bash:122]: This belongs in helpers/cli.sh

File: bats/helpers/funding/onchain.bash [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  exec_graphql "$token_name" 'on-chain-address-create' "$variables"
+  address="$(graphql_output '.data.onChainAddressCreate.address')"
+  [[ "${address}" != "null" ]] || exit 1
+
+  bitcoin_cli sendtoaddress "$address" "$btc_amount_in_btc"
+  bitcoin_cli -generate 2
+  retry 30 1 check_for_onchain_initiated_settled "$token_name" "$address"
+}
+
+check_for_onchain_initiated_settled() {
```
Comment (bodymindarts) [inline bats/helpers/funding/onchain.bash:23]: I don't think the `check_xx` and `get_from_transaction_by_address` functions belong here under `funding`.

File: bats/helpers/funding/wallet.bash [hunk 3]
```diff
@@ -0,0 +1,11 @@
+balance_for_wallet() {
```
Comment (bodymindarts) [inline bats/helpers/funding/wallet.bash:1]: not sure about having this under `funding`.
Funding (for me) implies sending / seeding money into a wallet. Getting a balance is a more general function.

What about something like this:
```
helpers
- onchain.bash
- intraledeger.bash
- ln.bash
- wallet.bash
```

Or something like that.

File: bats/helpers/funding/onchain.bash (hunk 2 above)
Comment (sandipndev) [reply bats/helpers/funding/onchain.bash]: FIxed in 9757bd4e56a6b4117e750b1782765b5ef35556e2

File: bats/helpers/funding/wallet.bash (hunk 3 above)
Comment (sandipndev) [reply bats/helpers/funding/wallet.bash]: fixed in 9757bd4e56a6b4117e750b1782765b5ef35556e2

**Review by bodymindarts — APPROVED** [review]

**Pull request merged**